robot/impl/resource_manager.go

```diff
@@ -683,6 +683,7 @@ func (manager *resourceManager) completeConfig(
 		// we use an errgroup here instead of a normal waitgroup to conveniently bubble
 		// up errors in resource processing goroutinues that warrant an early exit.
 		var levelErrG errgroup.Group
+		levelErrG.SetLimit(10)
```

Comment:
[nit] Could you just leave a comment above this line explaining why we're limiting to 10? (Something akin to your PR description.)